src/other/trace_tests/curl_tests/compare_expected_produced.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Define directories
+dir1="expected"
+dir2="produced"
+dir3="diffs"
+
+# Ensure that dir3 exists and is empty
+mkdir -p "$dir3"
+rm -f "$dir3"/*.diff
+
+# Loop over all files in dir1
+for file in "$dir1"/*; do
+    # Get the base name of the file (removing directory path)
+    basefile=$(basename "$file")
+
+    # Check if the file exists in dir2
+    if [[ -f "$dir2/$basefile" ]]; then
+        # If the file exists, run the git diff
+        diff_output=$(git diff --no-index "$dir1/$basefile" "$dir2/$basefile")
+
+        # Check if the diff output is non-empty
+        if [[ -n "$diff_output" ]]; then
+            # If it is non-empty, save it to a file in dir3
```

Comment:
Can you `echo` the corresponding `curl` command into the file here? (And into the expected results.) This would make it easy to copy and paste the curl command to the command line. It would create invalid JSON, so maybe not. Your call.